worksheet.rb

```diff
@@ -3,32 +3,157 @@
 # Which layers are "next" to each other?
+# id, date, cost, rider id and rating are next to each other
 
 ########################################################
 # Step 2: Assign a data structure to each layer
-
-# Copy your list from above, and in this section
-# determine what data structure each layer should have
+# - Layer 1: array of trips
+# - Layer 2: hash with key value pairs; where key is the header of the table and value is the data in a cell
 
 ########################################################
 # Step 3: Make the data structure!
+#
+rides = [
+    {
+        driver_id: "DR0004",
+        date: "3rd Feb 2016",
+        cost: 5,
+        rider_id: "RD0022",
+        rating: 5
+    },
+    {
+        driver_id: "DR0001",
+        date: "3rd Feb 2016",
+        cost: 10,
+        rider_id: "RD0003",
+        rating: 3
+    },
+    {
+        driver_id: "DR0002",
+        date: "3rd Feb 2016",
+        cost: 25,
+        rider_id: "RD0073",
+        rating: 5
+    },
+
+    {
+        driver_id: "DR0001",
+        date: "3rd Feb 2016",
+        cost: 30,
+        rider_id: "RD0015",
+        rating: 4
+    },
+
+    {
+        driver_id: "DR0003",
+        date: "4th Feb 2016",
+        cost: 5,
+        rider_id: "RD0066",
+        rating: 5
+    },
+
+    {
+        driver_id: "DR0004",
+        date: "4th Feb 2016",
+        cost: 10,
+        rider_id: "RD0022",
+        rating: 4
+    },
+
+    {
+        driver_id: "DR0002",
+        date: "4th Feb 2016",
+        cost: 15,
+        rider_id: "RD0013",
+        rating: 1
+    },
 
-# Setup the entire data structure:
-# based off of the notes you have above, create the
-# and manually write in data presented in rides.csv
-# You should be copying and pasting the literal data
-# into this data structure, such as "DR0004"
-# and "3rd Feb 2016" and "RD0022"
+    {
+        driver_id: "DR0003",
+        date: "5th Feb 2016",
+        cost: 50,
+        rider_id: "RD0003",
+        rating: 2
+    },
+
+    {
+        driver_id: "DR0002",
+        date: "5th Feb 2016",
+        cost: 35,
+        rider_id: "RD0066",
+        rating: 3
+    },
+
+    {
+        driver_id: "DR0004",
+        date: "5th Feb 2016",
+        cost: 20,
+        rider_id: "RD0073",
+        rating: 5
+    },
+
+    {
+        driver_id: "DR0001",
+        date: "5th Feb 2016",
+        cost: 45,
+        rider_id: "RD0003",
+        rating: 2
+    }
+]
 
 ########################################################
 # Step 4: Total Driver's Earnings and Number of Rides
-
 # Use an iteration blocks to print the following answers:
+
+def driver_info(rides, driver)
+  # creating the array of hashes with all rides for each driver
+  driver_rides = []
+  rides.each do |ride|
+    if ride[:driver_id] == driver
+      driver_rides << ride
+    end
+  end
+
+  #calculating the summary of costs of all rides for each driver
+  driver_earned = 0
+  driver_rides.each do |ride|
+    driver_earned += ride[:cost]
+  end
```

Comment:
Tip: This can be cleaned up using `sum`:

```suggestion
  driver_earned = driver_rides.sum { |ride| ride[:cost] }
```